### DISCUSSION TITLE
libuv timer repeat function set args error

### DISCUSSION THREAD

**Discussion by huangyidegit** [libuv/libuv#3805] [Q&A] [answered]
code at below,why uv_timer_start arg timeout use handle->repeat data,not handle->timeout?,is this a bug?:
int uv_timer_again(uv_timer_t* handle) {
  if (handle->timer_cb == NULL)
    return UV_EINVAL;

  if (handle->repeat) {
    uv_timer_stop(handle);
    uv_timer_start(handle, handle->timer_cb, handle->repeat, handle->repeat);
  }
  return 0;
}
libuv version:all

**Comment by huangyidegit** [comment]
i know now

**Comment by theanarkh** [comment, marked as answer]
Yeah. The first timeout is `handle->timeout`, and the next timeout is `handle->repeat`.

**Reply by huangyidegit** [reply]
thank you,i get it.
